### PR TITLE
chore: improve block namify 

### PR DESCRIPTION
**Pull request by naomi-lgbt**
Checklist:

- [X] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [X] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [X] My pull request targets the `main` branch of freeCodeCamp.
- [X] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

- Cleans up the preformatted names (we shouldn't need to preformat every block name that has HTML in it when we can preformat `HTML`)
- Removes edge case for `JavaScript`, in favour of `preformatted-words.json` to allow extension
- Adds `by` as a preposition it should not be title cased

This aims to solve some pain points I've seen in the review process for the new JavaScript projects. Leaving in draft to ensure I haven't broken anything (snapshots are being weird locally)

### CODE REVIEW

**Review by raisedadead — APPROVED** [review]
You write tests, you get approved.

File: utils/block-nameify.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import preFormattedBlockNames from './preformatted-block-names.json';
+import preFormattedWords from './preformatted-words.json';
 
-const noFormatting = ['and', 'for', 'of', 'the', 'up', 'with', 'to'];
+const noFormatting = ['and', 'for', 'of', 'the', 'up', 'with', 'to', 'by'];
```
Comment (ojeytonwilliams) [inline utils/block-nameify.ts:4]:
```suggestion
const noFormatting = ['and', 'for', 'of', 'the', 'up', 'with', 'to', 'by', 'a'];
```
I found in https://github.com/freeCodeCamp/freeCodeCamp/pull/49813/, that `'a'` was being uppercased, but it's more in this PR's scope than mine!

File: utils/block-nameify.test.ts [hunk 2]
```diff
@@ -6,9 +6,14 @@ describe('blockNameify', () => {
...
+  it('should use preformatted words when they exist', () => {
+    const result = blockNameify('html-css-javascript-fcc-freecodecamp');
+    expect(result).toBe('HTML CSS JavaScript fCC freeCodeCamp');
+  });
+
   it('should not format prepositions', () => {
-    const result = blockNameify('and-for-of-the-up-with');
-    expect(result).toBe('and for of the up with');
+    const result = blockNameify('and-for-of-the-up-with-by');
+    expect(result).toBe('and for of the up with by');
```
Comment (ojeytonwilliams) [inline utils/block-nameify.test.ts:16]:
```suggestion
    const result = blockNameify('and-for-of-the-up-with-by-a');
    expect(result).toBe('and for of the up with by a');
```

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]

**Review by ojeytonwilliams — APPROVED** [review]
LGTM :+1:

**Pull request merged**